components/server/src/workspace/workspace-service.ts

```diff
@@ -678,9 +683,13 @@ export class WorkspaceService {
     ): Promise<HeadlessLogUrls> {
         const workspace = await this.db.findByInstanceId(instanceId);
         if (!workspace) {
-            throw new ApplicationError(ErrorCodes.NOT_FOUND, `Workspace for instanceId ${instanceId} not found`);
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, `Prebuild for instanceId ${instanceId} not found`);
         }
-        await this.auth.checkPermissionOnWorkspace(userId, "access", workspace.id);
+        if (workspace.type !== "prebuild" || !workspace.projectId) {
+            throw new ApplicationError(ErrorCodes.CONFLICT, `Workspace is not a prebuild`);
```

Comment:
nit: Not sure Conflict is the correct error code. I always used BAD_REQUEST or METHOD_NOT_ALLOWED in these cases.